frontend/src/app/projects/create-project/create-project.component.ts

```diff
@@ -3,49 +3,149 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-// Copyright DB Netz AG and the capella-collab-manager contributors
-// SPDX-License-Identifier: Apache-2.0
-
-import { Component, OnInit } from '@angular/core';
-import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import {
+  AbstractControl,
+  FormControl,
+  FormGroup,
+  ValidationErrors,
+  Validators,
+} from '@angular/forms';
 import { MatStepper } from '@angular/material/stepper';
 import { Router } from '@angular/router';
-import { ProjectService } from 'src/app/services/project/project.service';
+import { ToastService } from '../../toast/toast.service';
+import {
+  BehaviorSubject,
+  connectable,
+  filter,
+  map,
+  Subject,
+  Subscription,
+  switchMap,
+  tap,
+} from 'rxjs';
+import slugify from 'slugify';
+import { NavBarService } from 'src/app/navbar/service/nav-bar.service';
+import {
+  Project,
+  ProjectService,
+} from 'src/app/services/project/project.service';
 
 @Component({
   selector: 'app-create-project',
   templateUrl: './create-project.component.html',
   styleUrls: ['./create-project.component.css'],
 })
-export class CreateProjectComponent implements OnInit {
-  nextLoading = false;
-  createProjectForm = new FormGroup({
+export class CreateProjectComponent implements OnInit, OnDestroy {
+  form = new FormGroup({
     name: new FormControl('', Validators.required),
     description: new FormControl(''),
-    editingMode: new FormControl('git'),
-    type: new FormControl('project'),
   });
 
-  project_slug = '';
+  private projectDetails = false;
+  private projectsSlugs = new BehaviorSubject<string[]>([]);
+  private slugsSubscription?: Subscription;
 
-  get name(): FormControl {
-    return this.createProjectForm.get('name') as FormControl;
+  constructor(
+    public projectService: ProjectService,
+    private router: Router,
+    private toastService: ToastService,
+    private navBarService: NavBarService
+  ) {
+    this.navBarService.title = 'Create Project';
   }
 
-  constructor(public projectService: ProjectService, private router: Router) {}
-
   ngOnInit(): void {
-    this.projectService.project = undefined;
+    this.form.controls.name.addValidators(this.slugValidator.bind(this));
```

Comment:
Is there a reason why you don't link the validator in the FormControl like the `required` validator? It makes it a lot more clear when they are all listed in one place (same applies for the `create-model.component.ts`).

Reply:
Yes, it’s because this validator relies indirectly on a constructor property, and hence cannot be fully known when added in the FormControl at definition. I was already aware of this issue but haven’t found a solution.

Reply:
The service that provides the current models value is provided by the modelService which is loaded by the constructor. Since it is an injectable, as far as I am aware it cannot be loaded elsewhere.